db/migrate/20190319192500_change_workflow_steps_default.rb

```diff
@@ -0,0 +1,12 @@
+class ChangeWorkflowStepsDefault < ActiveRecord::Migration
+  def change
+    reversible do |dir|
+      dir.up do
+        change_column_default(:workflows, :steps, [])
+        Workflow.find_each do |wf|
```

Comment:
I'd probably go with batch updating vs 1 sql update for every workflow, something like 
```
Workflow.select(:id).find_in_batches do |workflows| 
  workflows_update_scope =   Workflow.where(id: workflows.pluck(:id))
  workflows_update_scope.update_all(: steps, [])
end
```

That'll issue 1 update statement for every 1000 workflows so should be much faster.

Reply:
`find_each` does specifically uses batch processing via `find_in_batches`: https://apidock.com/rails/ActiveRecord/Batches/find_each

Reply:
It does indeed - the detail I raised was the `wf.update_column(:steps, [])` issues 1 sql update statement for each record all to set a default. If you are processing say 1000 records, that's 1000K individual SQL update statements (slow). using `find_in_batches` and `update_all` would issue 1 sql update statement.

Reply:
Oh...that one.